crates/rpc/rpc-types/src/eth/block.rs

```diff
@@ -49,20 +71,49 @@ pub struct Block {
 }
 
 impl Block {
-    /// Create a new block response from a [primitive block](reth_primitives::Block), using the
+    /// Converts the given primitive block into a [Block] response with the given
+    /// [BlockTransactionsKind]
+    pub fn from_block(
+        block: PrimitiveBlock,
+        total_difficulty: U256,
+        kind: BlockTransactionsKind,
+    ) -> Result<Self, BlockError> {
+        match kind {
+            BlockTransactionsKind::Hashes => {
+                Ok(Self::from_block_hashes_only(block, total_difficulty))
+            }
+            BlockTransactionsKind::Full => Self::from_block_full(block, total_difficulty),
+        }
+    }
+
+    /// Create a new [Block] response from a [primitive block](reth_primitives::Block), using the
     /// total difficulty to populate its field in the rpc response.
+    ///
+    /// This will populate the `transactions` field with only the hashes of the transactions in the
+    /// block: [BlockTransactions::Hashes]
+    pub fn from_block_hashes_only(block: PrimitiveBlock, total_difficulty: U256) -> Self {
+        let block_hash = block.header.hash_slow();
```

Comment:
is this intentional, because computing hash in place would be faster than db lookup?

Reply:
good point, I'll add support for providing the block hash directly in a followup